output/tar2/tar_output.go

```diff
@@ -31,28 +35,66 @@ func New(spec def.Output) *Output {
 	}
 }
 
-func (o Output) Apply(basePath string) <-chan error {
-	done := make(chan error)
+func (o Output) Apply(basePath string) <-chan output.Report {
+	done := make(chan output.Report)
 	go func() {
 		defer close(done)
 		try.Do(func() {
+			// open output location for writing
+			// currently this impl assumes a local file uri
+			file, err := os.OpenFile(o.spec.URI, os.O_CREATE|os.O_WRONLY|os.O_EXCL, 0755)
+			if err != nil {
+				done <- output.Report{errors.IOError.Wrap(err).(*errors.Error), def.Output{}}
```

Comment:
You could use a further anon closure that returns a Report and send that along. Shrug.